website/content/v1.10/kubernetes-guides/configuration/inlinemanifests.md

```diff
@@ -0,0 +1,138 @@
+    cluster:
+    extraManifests:
+        - "https://github.com/fluxcd/flux2/releases/latest/download/install.yaml"
+    EOF
+    ```
+
+1. Create a `CP_IPS` variable that contains the IP addresses of your control plane nodes:
+
+    ```bash
+    CP_IPS="<control-plane-ip-1>,<control-plane-ip-2>,<control-plane-ip-3>"
+    ```
+
+1. Run this command to export your `TALOSCONFIG` variable.
+You can skip this step if you've already done it:
+
+    ```bash
+    mkdir -p ~/.talos
+    cp ./talosconfig ~/.talos/config
+    ```
+
+1. Apply the `flux-extra-manifest.yaml` patch to your control plane nodes:
+
+    ```bash
+    talosctl patch machineconfig \
+    --patch @flux-extra-manifest.yaml \
+    --endpoints $CP_IPS \
+    --nodes $CP_IPS
+    ```
+
+1. Reboot the nodes.
+Note that if you have only one control plane node, rebooting it will cause cluster downtime.
+
+    ```bash
+    for NODE in $(echo "${CP_IPS}" | tr ',' ' '); do
+        echo "Rebooting control plane node: $NODE"
+        talosctl reboot --endpoints "$NODE" --nodes "$NODE" --wait
+    done
+    ```
+
+1. Wait a few seconds and check for the Flux pods:
+
+    ```bash
+    kubectl get pods -n flux-system -w
+    ```
+
+## Omni Patches
+
+You can also apply `inlineManifests` or `extraManifests` patches to Talos clusters managed by Omni.
+
+Refer to [Create a Patch For Cluster Machines](https://omni.siderolabs.com/how-to-guides/create-a-patch-for-cluster-machines?q=inline+manifest&ask=true) to learn how to create and apply the patches.
+
+## Summary: inlineManifests vs extraManifests
+
+Here’s a quick overview of the key differences between `inlineManifests` and `extraManifests`:
+
+|                        | `inlineManifests`                              | `extraManifests`                                             |
+| ---------------------- | ---------------------------------------------- | ------------------------------------------------------------ |
+| Source                 | Defined directly in the machine configuration  | Pulled from external URLs (GitHub gists, web servers, gists) |
+| Configuration Location | Under the `inlineManifests` section.           | Under the `extraManifests` section                           |
+| Usecase                | Early bootstrapping of critical resources      | For reusable, version-controlled, or shared manifests        |
+| Benefits               | No external dependencies                       | Centrally managed                                            |
+| Disadvantages          | Difficult to maintain and format embedded YAML | Requires external HTTP server                                |
+
+## Key Considerations
+
+One important thing to note is that Talos treats upgrades the same way it handles fresh installs.
+During an upgrade, Talos installs the new version onto a separate partition and then switches to it.
+As a result, any `inlineManifests` or `extraManifests` defined in the machine configuration will be re-applied.
```

Comment:
this is wrong. 

Talos logic is the following:

* it only creates missing Kubernetes resources
* Talos never updates or deletes Kubernetes resources

A sync for updated (but not removed) resources is described in https://www.talos.dev/v1.10/kubernetes-guides/upgrading-kubernetes/

Reply:
Oh my bad. Just to be sure, during upgrades:
1. Talos will create resources defined in your manifests if they do not exist in the cluster
2. Even if the resources exists, but has deviated from the manifest definition, Talos will leave the resources completely unchanged.
3. Talos WILL NOT delete resources from the cluster, even if they are no longer present in your manifest configuration.